### DISCUSSION TITLE
How do I download a single folder

### DISCUSSION THREAD

**Discussion by Aceplante** [community/community#139344] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Title. I'm just going to copy and paste my question from github since it was apparently too difficult for them.

I would like to download **a single folder** rather than each individual file in it and then needing to recreate the folder structure.

I have tried several extensions but they all seem to give me an error, perhaps becuase my repository is private?

How do I download a single folder from github (the website), preferably without all that command line stuff. Is that really the only way?

I would also like to import a folder TO github (the website). When I click and drag a folder from my local machine to the website, it converts it to a blank file with no data unless I am on the main repository level. If I navigate to a folder and then do the exact same thing it does not seem to import properly?

**Comment by sam-math** [comment, marked as answer]
Downloading and uploading folders directly from/to GitHub via the website interface can be tricky, especially with private repositories. Here's how to manage both tasks:

Downloading a Single Folder from GitHub (Without Command Line):

Unfortunately, GitHub does not provide a direct way to download an entire folder through the web interface. However, you can try some workarounds:

1. GitZip Extension: GitZip is a Chrome/Firefox extension that allows you to download a specific folder or file in GitHub. Since your repository is private, you need to ensure that you are logged into your GitHub account in the browser when using it. You can install GitZip from the Chrome/Firefox web store. If it still gives errors:

Make sure you're signed in to GitHub.

Confirm that your GitHub permissions for the extension are set up correctly to allow it to access private repositories.



2. GitHub Desktop (Optional): If you're open to using GitHub Desktop (which has a UI), you can clone your private repository, and after the repository is downloaded, navigate to the folder you want locally.

Download GitHub Desktop → Clone your private repo → Access the folder from your local machine.




If browser extensions don't work and you don't want to use GitHub Desktop, you may have to use the command line (git clone with sparse checkout), but you mentioned preferring not to.

Uploading a Folder to GitHub (Without Command Line):

1. Limitations of GitHub's Web Interface:

Main repository level: When you drag and drop a folder into the web interface at the root level (main page of the repository), it converts the folder into files.

Inside another folder: If you try to upload a folder inside another folder through the web UI, GitHub will not handle it correctly. It may create a blank file, as you noticed.



2. Workaround - Uploading Files Individually: If you prefer to avoid command-line tools:

Navigate to the folder in your repository where you want to upload files.

Instead of dragging an entire folder, drag and drop individual files within that folder.

Unfortunately, this is the only consistent way via the website interface.



3. Zip Folder: Alternatively, you can zip the folder on your local machine and upload the zipped file to GitHub. However, GitHub won't automatically extract it, so it's not ideal if you're looking to maintain a folder structure in the repository.



For both downloading and uploading folders, using GitHub Desktop or the command line is generally the most reliable method, especially for private repositories. However, these steps above might help depending on the complexity of your needs.

**Reply by Aceplante** [reply]
Thank you for being helpful! I did notice they seem to be website limitations, and I hoped I was just doing something wrong...

As for importing folders, if I recreate the folder structure and drag that into the repository level, the files go where I wanted them to.
